### PR TITLE
[TIMOB-20204] iOS: ListView - onload events of template items not fired

### PR DESCRIPTION
**Pull request by sriks**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-20204

[TIMOB-20204] iOS: ListView - onload events of template items not fired

### CODE REVIEW

File: iphone/Classes/TiUIImageViewProxy.m [hunk 1]
```diff
@@ -47,10 +51,34 @@ -(void)propagateLoadEvent:(NSString *)stateString
...
     }
+#ifdef TI_USE_KROLL_THREAD
+    else {
+        // Why do we do this?
+        // When running on kroll thread this is being called before the events are added.
+        // So we try to propagate this after the load event is added.
+        // TIMOB-20204
+        RELEASE_TO_NIL(self.loadEventState);
+        self.loadEventState = stateString;
+        self.modelDelegate = self;
```
Comment (hansemannn) [inline iphone/Classes/TiUIImageViewProxy.m]: Use Obj-C setters `[self setLoadEventState:stateString]`.

File: iphone/Classes/TiUIImageViewProxy.m [hunk 2]
```diff
@@ -47,10 +51,34 @@ -(void)propagateLoadEvent:(NSString *)stateString
...
+    }
+#endif
+}
+
+-(void)listenerAdded:(NSString*)type count:(int)count {
+    if ([self _hasListeners:@"load"]) {
+        self.modelDelegate = nil;
+        NSDictionary *event = [NSDictionary dictionaryWithObject:self.loadEventState forKey:@"state"];
```
Comment (hansemannn) [inline iphone/Classes/TiUIImageViewProxy.m]: You can use the Obj-C 2.0 syntax for Dictionaries here: 
```objc
[self fireEvent:@"load" withObject:@{@"state": [self loadEventState]}];
```

File: iphone/Classes/TiUIImageViewProxy.m [hunk 3]
```diff
@@ -47,10 +51,34 @@ -(void)propagateLoadEvent:(NSString *)stateString
...
+	
     if ([self _hasListeners:@"load"]) {
         NSDictionary *event = [NSDictionary dictionaryWithObject:stateString forKey:@"state"];
         [self fireEvent:@"load" withObject:event];
     }
+#ifdef TI_USE_KROLL_THREAD
+    else {
+        // Why do we do this?
+        // When running on kroll thread this is being called before the events are added.
+        // So we try to propagate this after the load event is added.
```
Comment (hansemannn) [inline iphone/Classes/TiUIImageViewProxy.m]: Does ist work properly with main-thread enabled (default in the Xcode project)?

File: iphone/Classes/TiUIImageViewProxy.m (hunk 1 above)
Comment (sriks) [inline iphone/Classes/TiUIImageViewProxy.m]: Done

File: iphone/Classes/TiUIImageViewProxy.m (hunk 2 above)
Comment (sriks) [inline iphone/Classes/TiUIImageViewProxy.m]: Done

File: iphone/Classes/TiUIImageViewProxy.m (hunk 3 above)
Comment (sriks) [inline iphone/Classes/TiUIImageViewProxy.m]: @hansemannn 
I tested with TI_USE_KROLL_THREAD enabled/disabled in defines.h. But I strongly suggest you to try it as well. Always good to have a different set of eyes!

**Comment by hansemannn** [comment]
Did some integration test with- and without the listView and `load` event. Works fine! PR approved! :rocket:

**Pull request merged**